harness/determined/cli/dev.py

```diff
@@ -0,0 +1,83 @@
+import json
+import shlex
+import shutil
+import subprocess
+import sys
+from argparse import Namespace
+from typing import Any, List
+
+from termcolor import colored
+
+from determined.common.api import authentication
+from determined.common.api.request import make_url
+from determined.common.declarative_argparse import Arg, Cmd
+
+
+@authentication.required
+def token(_: Namespace) -> None:
+    token = authentication.must_cli_auth().get_session_token()
+    print(token)
+
+
+@authentication.required
+def curl(args: Namespace) -> None:
+    assert authentication.cli_auth is not None
+    if shutil.which("curl") is None:
+        print(colored("curl is not installed on this machine", "red"))
+        sys.exit(1)
+
+    cmd: List[str] = [
+        "curl",
+        make_url(args.master, args.path),
+        "-H",
+        f"Authorization: Bearer {authentication.cli_auth.get_session_token()}",
+        "-s",
+    ]
+    if args.curl_args:
+        cmd.append(args.curl_args)
+
+    if args.x:
+        if hasattr(shlex, "join"):  # added in py 3.8
+            print(shlex.join(cmd))  # type: ignore
+        else:
+            print(" ".join(shlex.quote(arg) for arg in cmd))
+    output = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    if output.stderr:
+        print(output.stderr.decode("utf8"), file=sys.stderr)
+
+    out = output.stdout.decode("utf8")
+    try:
+        json_resp = json.loads(out)
+        if shutil.which("jq") is not None:
+            subprocess.run(["jq", "."], input=out, text=True)
+        else:
+            print(json.dumps(json_resp, indent=4))
+    except json.decoder.JSONDecodeError:
+        print(out)
+
```

Comment:
this is basically an opportunistic prettyprinting. it's not always possible to pip the det cli output out to say jq as it might be printing other messages eg for user login or version update warnings.

Reply:
we could probably use some shared formatter for similar purpose to print yaml, json, etc wherever we do these eg

https://github.com/determined-ai/determined/blob/9beebcc1ac86041c806c121e9bbfc33f7d14e373/harness/determined/cli/workspace.py#L117

https://github.com/determined-ai/determined/blob/9beebcc1ac86041c806c121e9bbfc33f7d14e373/harness/determined/cli/_util.py#L14

Reply:
we have a ton of individual similar calls `ag 'dumps.*indent' harness/determined/cli/ -l`